From 24839df617aef8099e4d1ba3fc47a04c52c0cf75 Mon Sep 17 00:00:00 2001
From: Justin Bronder <jsbronder@cold-front.org>
Date: Wed, 6 Nov 2024 11:59:40 -0500
Subject: [PATCH] lib: lz4: use lz4 instead of lz4c

lz4c has been considered deprecated by upstream since at least 2018 [1]
and has been disabled by default recently [2].  openSUSE Tumbleweed is
no longer packaging the deprecated version and others will likely
follow.

Going back as far as Ubuntu 16.04/Fedora 25, both lz4 and lz4c are
installed as part of the same package so switching should be safe.

1. https://github.com/lz4/lz4/pull/553
2. https://github.com/lz4/lz4/pull/1479

Upstream-Status: Submitted [http://lists.infradead.org/pipermail/barebox/2024-November/048491.html]
Signed-off-by: Justin Bronder <jsbronder@cold-front.org>
---
 scripts/Makefile.lib | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index 1881e3c1..1052bcee 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -492,7 +492,7 @@ cmd_lzo = (cat $(filter-out FORCE,$^) | \
 
 quiet_cmd_lz4 = LZ4     $@
 cmd_lz4 = (cat $(filter-out FORCE,$^) | \
-	lz4c -l -c1 stdin stdout && $(call size_append, $(filter-out FORCE,$^))) > $@ || \
+	lz4 -l --best stdin stdout && $(call size_append, $(filter-out FORCE,$^))) > $@ || \
 	(rm -f $@ ; false)
 
 %.lz4: %
